src/components/MapView/DateSelector/utils.ts

```diff
@@ -60,3 +60,48 @@ export function isAvailableMonth(
 ) {
   return findAvailableDayInMonth(month, year, availableDates).length > 0;
 }
+
+/**
+ * Return index of available dates that matched
+ * @param availableDates in millisecond format
+ * @param date in millisecond format
+ * @return
+ */
+export function findDateIndex(
```

Comment:
This *looks* like a binary search algorithm modified to find the closest match, but not too sure. If it is, are we assuming that availableDates is sorted? Might be worthwhile to document if so.

Reply:
you are right, it is a binary search to find closest match. i updated the documentation